R/standard-data-prep/utils.R

```diff
@@ -0,0 +1,177 @@
+    new_lookup <- new_lookup %>% select(-ObjectId)
+  }
+
+  # Extra tidy up for separating LAs / LADs
+  # This bit's to deal with LAs and UTLAs not being quite the same thing. 
+  # The parent UTLA for LADs in metropolitan counties (E11, e.g. Manchester),
+  # and Inner or Outer London (E13) are the Metropolitan counties 
+  # and Inner / Outer London. 
+  # But in these cases, the LA for our purposes is the LAD itself.
+  
+  # So this is using the LAD as it's own parent LA if it's in a metropolitan
+  # county or in London and taking the UTLA otherwise.
+  
+  So all this logic is doing is using the LAD as it's own parent LA if it's in a metropolitan county or in London and taking the UTLA otherwise.
+  if ("new_la_code" %in% names(new_lookup)) {
+    new_lookup1 <- new_lookup %>%
+      mutate(
+        la_name = if_else(
+          grepl("E11", new_la_code) | grepl("E13", new_la_code),
+          lad_name,
+          la_name
+        ),
+        new_la_code = if_else(
+          grepl("E11", new_la_code) | grepl("E13", new_la_code),
+          lad_code,
+          new_la_code
+        )
+      )
+  }
+
+  return(new_lookup)
+}
+
+#' Overwrite the existing lookup file by appending new data
+#'
+#' Take a tidied file, likely produced by the tidy_downloaded_lookup function
+#' and either append to existing lookup file or write a new lookup file if
+#' one doesn't exist at the given file path
+#'
+#' If a lookup file already exists it will automatically only use the columns
+#' present in that file
+#'
+#' @param new_lookup data frame of new lookup table,
+#' usually the output of tidy_downloaded_lookup
+#' @param lookup_filepath file path to existing lookup you want to update,
+#' or the path for the new lookup to be written to
+#'
+#' @return message confirming if the CSV file has been successfully written
+
+write_updated_lookup <- function(
+    new_lookup,
+    lookup_filepath) {
+  # If a lookup file exists already read it in
+  if (file.exists(lookup_filepath)) {
+    message("Reading data from existing file: ", lookup_filepath)
+    existing_lookup <- read_csv(lookup_filepath, show_col_types = FALSE)
+
+    # Filter the new lookup to only have the same columns
+    new_lookup <- new_lookup %>% select(all_of(names(existing_lookup)))
+  } else {
+    existing_lookup <- data.frame()
+    message("No existing lookup found at: ", lookup_filepath)
+  }
+
+  # Work out the geography columns we care about in the new lookup
+  cols_to_join_by <- names(new_lookup)[!names(new_lookup) %in% c("first_available_year_included", "most_recent_year_included")]
+
+  # Append the new lookup
+  updated_lookup <- existing_lookup %>%
+    rbind(new_lookup) %>%
+    # Then condense the rows, rewriting the first and last years for each row
+    summarise(
+      first_available_year_included = min(first_available_year_included),
+      most_recent_year_included = max(most_recent_year_included),
+      .by = all_of(cols_to_join_by)
+    )
```

Comment:
I'd quite like a bit of clean up here using `arrange()`.

I'm thinking maybe something along the lines of:
```
%>% arrange(names(new_lookup)[names(new_lookup) %in% c("ward_code", "pcon_code", "lad_code", "new_la_code", "lsip_code")])
```

Reply:
What's the clean up you'd want to see?

I'm not sure we'd want to be as specific in this function? My thinking was this was fairly generalised so that any columns could be passed through and then if you wanted to filter any out, that's easy enough to do before you run the 'write' function?

Reply:
We spoke about this, will add some generic ordering so it's done by year available and the code column

Reply:
Added now